src/assets/TokenRatesController.ts

```diff
@@ -196,8 +197,8 @@ export class TokenRatesController extends BaseController<
     };
     this.initialize();
     this.configure({ disabled: false }, false, false);
-    onTokensStateChange((tokensState) => {
-      this.configure({ tokens: tokensState.tokens });
+    onTokensStateChange(({ tokens, detectedTokens }) => {
+      this.configure({ tokens: [...tokens, ...detectedTokens] });
```

Comment:
It would be great to capture this change in a unit test. It doesn't look like any of them test for whether rates for detected tokens are collected or not.

Reply:
Likewise for this same change in the token balances controller

Reply:
Added test to check for this